pkg/cloudclient/gcp/private.go

```diff
@@ -0,0 +1,417 @@
+	handledErrors "github.com/openshift/osd-network-verifier/pkg/errors"
+	"github.com/openshift/osd-network-verifier/pkg/helpers"
+	"github.com/openshift/osd-network-verifier/pkg/output"
+)
+
+type createComputeServiceInstanceInput struct {
+	ContOptImageID string
+	vpcSubnetID    string
+	userdata       string
+	// ebsKmsKeyID   string
+	zone         string
+	machineType  string
+	instanceName string
+	sourceImage  string
+	networkName  string
+}
+
+// //global variable ContOptImage image, gcp has region, zone
+
+var (
+	defaultContOptImage = map[string]string{
+		// using google container optimized image
+		"default": "cos-97-lts",
+		//other regions to add
+	}
+	// TODO find a location for future docker images
+	networkValidatorImage string = "quay.io/app-sre/osd-network-verifier:v0.1.159-9a6e0eb"
+	userdataEndVerifier   string = "USERDATA END"
+)
+
+//newClient method
+func newClient(ctx context.Context, logger ocmlog.Logger, credentials *google.Credentials, region, instanceType string, tags map[string]string) (*Client, error) {
+	//use oauth2 token in credentials struct to create client,
+	// https://pkg.go.dev/golang.org/x/oauth2/google#Credentials
+
+	// https://cloud.google.com/docs/authentication/production
+	//service account credentials order - env variable, service account attached to resource, error
+
+	computeService, err := computev1.NewService(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	c := &Client{
+		projectID:      credentials.ProjectID,
+		region:         region,
+		zone:           fmt.Sprintf("%s-b", region), //append zone b
+		instanceType:   instanceType,
+		computeService: computeService,
+		tags:           tags,
+		logger:         logger,
+		output:         output.Output{},
+	}
+
+	if err := c.validateMachineType(ctx); err != nil {
+		return nil, fmt.Errorf("Instance type %s is invalid: %v", c.instanceType, err)
+	}
+
+	return c, nil
+}
+
+func (c *Client) validateMachineType(ctx context.Context) error {
+	//  machineTypes List https://cloud.google.com/compute/docs/reference/rest/v1/machineTypes/list
+
+	c.logger.Debug(ctx, "Gathering description of instance type %s from EC2", c.instanceType)
+
+	descOut := c.computeService.MachineTypes.List(c.projectID, c.zone)
+
+	//check if instanceType is in the list
+	found := false
+	if err := descOut.Pages(ctx, func(page *computev1.MachineTypeList) error {
+		for _, machineType := range page.Items {
+			if string(machineType.Name) == c.instanceType {
+				found = true
+				c.logger.Debug(ctx, "Instance type %s supported", c.instanceType)
+				break
+			}
+		}
+		c.logger.Debug(ctx, "Fully describe instance types output contains %d instance types", len(page.Items))
+		return nil
+	}); err != nil {
+		return fmt.Errorf("Unable to gather list of supported instance types from ComputeService: %v", err)
+	}
+
+	if !found {
+		return fmt.Errorf("Instance type %s not found in ComputeService API", c.instanceType)
+	}
+
+	return nil
+}
+
+//ToDo func createComputeServiceInstance
```

Comment:
Comments likely need to be reviewed/cleaned up

Reply:
Done!